ibmcloud/image/build.sh

```diff
@@ -206,69 +257,157 @@ umount "$src_mnt/dev"
+    echo -e "\nFormatting root partition"
+    mkfs.ext4 -L "$label" /dev/mapper/$LUKS_NAME
+    mount /dev/mapper/$LUKS_NAME "$dst_mnt"
+    mkdir ${dst_mnt}/etc
+    mkdir ${dst_mnt}/boot-se
+    mount -o norecovery ${tmp_nbd}p1 ${dst_mnt}/boot-se
+else
+    label=$(lsblk -n -o label "$src_part")
+    mkfs.ext4 -L "$label" "$dst_part"
+    mount "$dst_part" "$dst_mnt"
+fi
 
+echo "Copying the root filesystem"
 tar_opts=(--numeric-owner --preserve-permissions --acl --selinux --xattrs --xattrs-include='*' --sparse)
 tar -cf - "${tar_opts[@]}" --sort=none -C "$src_mnt" . | tar -xf - "${tar_opts[@]}" --preserve-order  -C "$dst_mnt"
 
+echo "The root filesystem is ready"
+sleep 1
 umount "$src_mnt"
-
 mount -t sysfs sysfs "$dst_mnt/sys"
 mount -t proc proc "$dst_mnt/proc"
 mount --bind /dev "$dst_mnt/dev"
 
-case "$(uname -m)" in
-    s390x)
-        echo -e "\nExecuting zipl\n"
-        helper="$dst_mnt/lib/s390-tools/zipl_helper.nbd"
-        cat <<END > "$helper"
+if [ "${SE_BOOT}" = "1" ]; then
+    echo -e "\nPreparing secure execution boot image"
+    echo "mounting tmpfs to /etc/keys"
+    mkdir -p ${dst_mnt}/etc/keys
+    mount -t tmpfs keys ${dst_mnt}/etc/keys
+    # ADD CONFIGURATION
+    echo "adding fstab"
+    echo "#This file was auto-generated" >  ${dst_mnt}/etc/fstab
+    echo "/dev/mapper/$LUKS_NAME    /        ext4  defaults 1 1" >>  ${dst_mnt}/etc/fstab
+    echo "PARTUUID=$BOOT_UUID    /boot-se    ext4  norecovery 1 2" >> ${dst_mnt}/etc/fstab
+    echo "adding luks keyfile for fs"
+    cp rootkey.bin ${dst_mnt}/etc/keys/luks-$(blkid -s UUID -o value /dev/mapper/$LUKS_NAME).key
+    chmod 600 ${dst_mnt}/etc/keys/luks-$(blkid -s UUID -o value /dev/mapper/$LUKS_NAME).key
+    echo "#This file was auto-generated" >  ${dst_mnt}/etc/crypttab
+    echo "$LUKS_NAME UUID=$(blkid -s UUID -o value ${dst_part}) /etc/keys/luks-$(blkid -s UUID -o value /dev/mapper/$LUKS_NAME).key luks,discard,initramfs" >> ${dst_mnt}/etc/crypttab
+    chmod 744 ${dst_mnt}/etc/crypttab
+    # Disable virtio_rng
+    echo "#don't trust rng from hypervisor"  > ${dst_mnt}/etc/modprobe.d/blacklist-virtio.conf
+    echo "blacklist virtio_rng"             >> ${dst_mnt}/etc/modprobe.d/blacklist-virtio.conf
+
+    # Favor loading of TRNG module for newer Z machines
+    echo 's390_trng' >> ${dst_mnt}/etc/modules
+
+    # Prep files needed for mkinitrd (if running encrypted /)
+    echo "KEYFILE_PATTERN=\"/etc/keys/*.key\"" >> ${dst_mnt}/etc/cryptsetup-initramfs/conf-hook
+    echo "UMASK=0077" >> ${dst_mnt}/etc/initramfs-tools/initramfs.conf
+    cat <<END > ${dst_mnt}/etc/zipl.conf
+[defaultboot]
+default=linux
+target=/boot-se
+
+targetbase=/dev/vda
+targettype=scsi
+targetblocksize=512
+targetoffset=2048
+
+[linux]
+image = /boot-se/se.img
+END
+    echo "updating initial ram disk"
+    chroot "$dst_mnt" update-initramfs -u || echo -e "!!! Bootloader install errors prior to this line are intentional !!!!!"
+    echo -e "\nGenerating an IBM Secure Execution image"
```

Comment:
nit: instead of `echo -e`, you can also use `printf` built-in (does not require `-e`)

Reply:
updated.